hack/.deployUtils

```diff
@@ -1,67 +1,10 @@
-#!/bin/bash
-
-#
-# Copyright 2022 Red Hat, Inc.
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#     http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-#
+# shellcheck shell=bash
 
-LOCAL_SETUP_DIR="$(dirname "${BASH_SOURCE[0]}")"
-source "${LOCAL_SETUP_DIR}"/.setupEnv
-source "${LOCAL_SETUP_DIR}"/.kindUtils
-source "${LOCAL_SETUP_DIR}"/.clusterUtils
-source "${LOCAL_SETUP_DIR}"/.argocdUtils
-source "${LOCAL_SETUP_DIR}"/.cleanupUtils
-
-KIND_CLUSTER_PREFIX="mgc-"
-KIND_CLUSTER_CONTROL_PLANE="${KIND_CLUSTER_PREFIX}control-plane"
-KIND_CLUSTER_WORKLOAD="${KIND_CLUSTER_PREFIX}workload"
-
-INGRESS_NGINX_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/ingress-nginx
-METALLB_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/metallb
-CERT_MANAGER_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/cert-manager
-EXTERNAL_DNS_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/external-dns
-ARGOCD_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/argocd
-ISTIO_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/istio/istio-operator.yaml
-GATEWAY_API_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/gateway-api
-REDIS_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/kuadrant/redis
-LIMITADOR_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/kuadrant/limitador
-THANOS_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/thanos
-PROMETHEUS_FOR_FEDERATION_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/prometheus-for-federation
-GRAFANA_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/grafana
-
-TLS_CERT_PATH=${LOCAL_SETUP_DIR}/../config/webhook-setup/control/tls
-
-set -e pipefail
-
-deployIngressController () {
-  clusterName=${1}
-  kubectl config use-context kind-${clusterName}
-  echo "Deploying Ingress controller to ${clusterName}"
-  ${KUSTOMIZE_BIN} build ${INGRESS_NGINX_KUSTOMIZATION_DIR} --enable-helm --helm-command ${HELM_BIN} | kubectl apply -f -
-  echo "Waiting for deployments to be ready ..."
-  kubectl -n ingress-nginx wait --timeout=600s --for=condition=Available deployments --all
-}
 
-deployMetalLB () {
-  clusterName=${1}
-  metalLBSubnet=${2}
+# Shared functions between local-setup-mgc and quickstart-setup script
 
-  kubectl config use-context kind-${clusterName}
-  echo "Deploying MetalLB to ${clusterName}"
-  ${KUSTOMIZE_BIN} build ${METALLB_KUSTOMIZATION_DIR} | kubectl apply -f -
-  echo "Waiting for deployments to be ready ..."
-  kubectl -n metallb-system wait --for=condition=ready pod --selector=app=metallb --timeout=300s
+configureMetalLB () {
+  metalLBSubnet=${1}
```

Comment:
```suggestion
  clusterName=${1}
  metalLBSubnet=${2}

  kubectl config use-context kind-${clusterName}
  echo "Deploying MetalLB to ${clusterName}"}
```

Reply:
is there a use for the cluster name? I'd just remove it - one less var to swing around

Reply:
it's used to set the context inside the function. if we don't use it the change should still include metalLBSubent=${2} otherwise the wrong value is used in the function call from the quickstart.

Reply:
I'm a bit confused. There is a [deployMetaLB](https://github.com/Kuadrant/multicluster-gateway-controller/blob/main/hack/local-setup.sh#L56-L82) that is used only for the local setup. And there is a [configureMetaLB](https://github.com/Kuadrant/multicluster-gateway-controller/blob/main/hack/quickstart-setup.sh#L39-L58) that is used in the [quickstart](https://github.com/Kuadrant/multicluster-gateway-controller/blob/main/hack/quickstart-setup.sh#L273) and is called from the [deployMetaLB](https://github.com/Kuadrant/multicluster-gateway-controller/blob/fa90bdef31b8234dfc5b27fe366cf3d98b54b84e/hack/.deployUtils#L114) function. The configure one is not using context as in case it is called from the deployMetaLB the contexts is already set [here](https://github.com/Kuadrant/multicluster-gateway-controller/blob/fa90bdef31b8234dfc5b27fe366cf3d98b54b84e/hack/.deployUtils#L109) and if it is called from the quickstart script the function before it [sets](https://github.com/Kuadrant/multicluster-gateway-controller/blob/main/hack/quickstart-setup.sh#L126) the context. I see how it can break, so would you suggest to actually use the passed inside clusterName to ensure the contexts was set? :thinking:

Reply:
we moved it to configureMetalLB because we moved the deployment of metalLB to kustomize for the quickstart, you could move the local setup to install by kustomize as well and that would bring the two approaches to the same. even if we don't do that, both calls to the function should call in the same way. e.g. remove the passing of the context value from the quickstart if it's not needed. although i think safer that it is. so pass the context from the call inside deployMetalLB

Reply:
Yeah, better to be extra safe. Passed around `clusterName` for extra safety